tmt/steps/provision/__init__.py

```diff
@@ -260,6 +263,9 @@ def __init__(self, data, name=None, parent=None):
         self.role = None
         # Flag to indicate localhost guest, requires special handling
         self.localhost = False
+        # Flag to indicate guest is ready or not. It is true by default, if any
+        # error occurs, should set it to be False
+        self.is_ready = True
```

Comment:
I for one would prefer starting with `False` as the default: there is probably a single point in each provisioning plugin after which the guest becomes ready, while there are many places where `tmt` code flow leaves guest provisioning, failing, and must mark guest as "not ready" - and it's simply very easy to forget to do so. Therefore starting with "not ready" and changing to "ready" at points where we are sure the guest is indeed up & running would mean less space for omission plus fewer places that need to be touched by the patch in general.

Also, it would be consistent for an outside observer: on [this](https://github.com/teemtee/tmt/blob/main/tmt/steps/provision/artemis.py#L434) line, after creating `GuestArtemis` brand new instance, `self._guest.is_ready` would read `True`, which is simply incorrect - several API calls must be made first to actually provision the VM before `is_ready` could read `True`.